# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContextHelper.cpp
# Commit: cd4ca1691c0b
# Full Hash: cd4ca1691c0b74f665e4e43692d04a992d2a11db
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2022-06-23 03:54:50
# Regressor Bug: 1769878
# File Overlap Count: 1
# Description:
#   Bug 1769878, ensure objects owned by the worker thread are cleared when the worker is shutting down, r=asuth,aosmond
#   
#   Depends on D147858
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D149693
# ==============================================================================

diff -r 851d0144fd91 -r cd4ca1691c0b dom/canvas/CanvasRenderingContextHelper.cpp
--- a/dom/canvas/CanvasRenderingContextHelper.cpp	Wed Jun 22 14:06:30 2022 +0000
+++ b/dom/canvas/CanvasRenderingContextHelper.cpp	Wed Jun 22 14:06:31 2022 +0000
@@ -39,6 +39,8 @@
     // This is called on main thread.
     MOZ_CAN_RUN_SCRIPT
     nsresult ReceiveBlobImpl(already_AddRefed<BlobImpl> aBlobImpl) override {
+      MOZ_ASSERT(NS_IsMainThread());
+
       RefPtr<BlobImpl> blobImpl = aBlobImpl;
 
       RefPtr<Blob> blob;
@@ -58,6 +60,11 @@
       return rv.StealNSResult();
     }
 
+    bool CanBeDeletedOnAnyThread() override {
+      // EncodeCallback is used from the main thread only.
+      return false;
+    }
+
     nsCOMPtr<nsIGlobalObject> mGlobal;
     RefPtr<BlobCallback> mBlobCallback;
   };